# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/binutils/i686-nopl.patch
# Copyright (C) 2011 - 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Despite months of prominent people arguing, this again goes thru half fixed,
sigh!

By default GCC (and glibc Makefiles) do not pass -march= down to the
assembler, and -mtune=i686 only let the UNKNOWN march path choose the
wrong nopl from the i686 marked PENTIUM_PRO entry, ... I choose to just
touch the "march" not selected case, as changing the i686 entry to
GENERIC32, or so, may introduce other regressions related to CMOV, et al.

  - Rene Rebe <rene@exactcode.de>

--- binutils-2.47/gas/config/tc-i386.c.vanilla	2026-07-27 12:23:28.233573106 +0200
+++ binutils-2.47/gas/config/tc-i386.c	2026-07-27 12:25:29.539676459 +0200
@@ -1640,13 +1640,6 @@
 	     explicitly disabled.  */
 	  switch (fragP->tc_frag_data.tune)
 	    {
-	    case PROCESSOR_UNKNOWN:
-	      /* We use cpu_arch_isa_flags to check if we SHOULD
-		 optimize with nops.  */
-	      if (fragP->tc_frag_data.isanop)
-		patt = alt_patt;
-	      break;
-
 	    case PROCESSOR_CORE:
 	    case PROCESSOR_CORE2:
 	    case PROCESSOR_COREI7:
@@ -1680,6 +1673,7 @@
 	    case PROCESSOR_I686:
 	    case PROCESSOR_IAMCU:
 	    case PROCESSOR_GENERIC32:
+	    case PROCESSOR_UNKNOWN:
 	      break;
 	    case PROCESSOR_NONE:
 	      abort ();
